control-plane/api-gateway/translation/k8s_cache_translation.go

```diff
@@ -0,0 +1,131 @@
+package translation
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/hashicorp/consul/api"
+)
+
+type TranslatorFn func(api.ConfigEntry) []types.NamespacedName
+
+type secretTransfomer func(context.Context) func(client.Object) []reconcile.Request
+
+type resourceGetter interface {
+	Get(api.ResourceReference) api.ConfigEntry
+}
+
+// ConsulToNSNTranslator handles translating consul config entries to k8s namespaced names
+type ConsulToNSNTranslator struct {
```

Comment:
```suggestion
type ConsulToNamespacedNameTranslator struct {
```
Takes longer to read, but saved me a few seconds of furrowing my brows to figure out what NSN was in this context. That will help keep my wrinkle-free for the years to come.

Reply:
good call!